test/support/common.go

```diff
@@ -0,0 +1,177 @@
+package support
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/image"
+	"github.com/docker/docker/client"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func GetEnvOrDefault(key, defaultValue string) string {
+	return getEnvOrDefault(key, defaultValue, true)
+}
+
+func GetEnvOrDefaultSecret(key, defaultValue string) string {
+	return getEnvOrDefault(key, defaultValue, false)
+}
+
+func getEnvOrDefault(key, defaultValue string, isLogged bool) string {
+	var returnValue string
+	isDefaultValue := false
+	value, exists := os.LookupEnv(key)
+	if !exists && defaultValue != "" {
+		returnValue = defaultValue
+		isDefaultValue = true
+	} else {
+		returnValue = value
+	}
+	var logMessage string
+	if isLogged || returnValue == "" {
+		logMessage = fmt.Sprintf("%s='%s'", key, returnValue)
+	} else {
+		logMessage = fmt.Sprintf("%s=%s", key, "*****")
+	}
+	if isDefaultValue {
+		logMessage = fmt.Sprintf("%s (default)", logMessage)
+	}
+	log.Println(logMessage)
+	return returnValue
+}
+
+func GetFileContent(filePath string) (string, error) {
+	snapshotFile, isLocal := checkFilePath(filePath)
+	if isLocal {
+		return loadFileContent(snapshotFile)
+	} else {
+		githubToken := GetEnvOrDefaultSecret(EnvTestGithubToken, "")
+		return downloadFileContent(snapshotFile, githubToken)
+	}
+}
+
+func checkFilePath(filePath string) (string, bool) {
+	if strings.HasPrefix(filePath, "http://") || strings.HasPrefix(filePath, "https://") {
+		return filePath, false
+	} else {
+		filePath = localPathCleanup(filePath)
+		return filePath, true
+	}
+}
+
+func localPathCleanup(origPath string) string {
+	finalPath := origPath
+	if !filepath.IsAbs(origPath) {
+		// not ideal solution
+		// want to have path relative to the project directory
+		// without test/acceptance-tests
+		finalPath = filepath.Join("..", "..", origPath)
```

Comment:
check this https://stackoverflow.com/questions/31873396/is-it-possible-to-get-the-current-root-of-package-structure-as-a-string-in-golan

Reply:
Great tip. Thanks, I used it :-)